test/iteration.jl

```diff
@@ -22,13 +22,23 @@ using Test, DataFrames
         @test collect(pairs(row)) isa Vector{Pair{Symbol, Int}}
     end
 
-    @test size(eachcol(df)) == (size(df, 2),)
+    @test Base.IteratorSize(eachcol(df)) == Base.HasShape{1}()
     @test parent(eachcol(df)) === df
     @test names(eachcol(df)) == names(df)
-    @test IndexStyle(eachcol(df)) == IndexLinear()
-    @test Base.IndexStyle(eachcol(df)) == IndexLinear()
     @test length(eachcol(df)) == size(df, 2)
+    @test size(eachcol(df)) == (size(df, 2),)
+    @test size(eachcol(df), 1) == size(df, 2)
+    @test size(eachcol(df), 2) == 1
+    @test_throws ArgumentError size(eachcol(df), 0)
     @test eachcol(df)[1] == df[:, 1]
+    @test eachcol(df)[:A] == df[:, :A]
+    @test eachcol(df)[All()] == eachcol(df)
+    @test isequal(eachcol(df)[[1]], eachcol(df[!, [1]]))
+    @test eachcol(df).A == df[:, :A]
+    @test eachcol(df)["A"] == df[:, "A"]
+    @test eachcol(df)."A" == df[:, "A"]
+    @test eachcol(df)[begin] == df[!, 1]
```

Comment:
I think you need to wrap this test in a version check for whenever `begin` was allowed in indexing (tests for 1.0 fail because of it)

Reply:
Indeed. Actually we have a separate file for these tests (as we have to conditionally `include` it), but I have forgotten about it :disappointed:.